gui/journal.lua

```diff
@@ -1,10 +1,80 @@
 -- Fort journal with a multi-line text editor
+--@ module = true
 
 local gui = require 'gui'
 local widgets = require 'gui.widgets'
 
 local CLIPBOARD_MODE = {LOCAL = 1, LINE = 2}
 
+-- This class caches lines of text wrapped to a specified width for performance
+-- and readability. It can convert a given text index to (x, y) coordinates in
+-- the wrapped text and vice versa.
+
+-- Usage:
+-- This class should only be used in the following scenarios.
+--  1. When text or text features need to be rendered
+--     (wrapped {x, y} coordinates are required).
+--  2. When mouse input needs to be converted to the original text position.
+
+-- Using this class in other scenarios may lead to issues with the component's
+-- behavior when the text is wrapped.
+WrappedText = defclass(WrappedText)
+
+WrappedText.ATTRS{
+    text = '',
+    wrap_width = 0,
```

Comment:
`0` isn't a valid width to pass to `string:wrap`; you might as well use `wrap_width = DEFAULT_NIL,` here